harmony_netcdf_to_zarr/mosaic_utilities.py

```diff
@@ -241,24 +268,107 @@ def _get_temporal_output_dimension(self,
+            non-consecutive input granules). If there are missing bounds values
+            for the output dimension, use the median difference between the
+            input bounds and dimension values to derive the missing values.
+
+            If the input dimension data did not have an associated bounds
+            variable, this method with return `None` for both the values and
+            path of the bounds.
+
+        """
+        bounds_path = next(
+            iter(self.input_dimensions[dimension_name].values())
+        ).bounds_path
+
+        if bounds_path is not None:
+            bounds_values = np.zeros((output_dimension_values.size, 2),
+                                     dtype=output_dimension_values.dtype)
+            filled_indices = set()
+
+            for input_dimension in self.input_dimensions[dimension_name].values():
+                # For each granule, match the dimension values to the output
+                # dimension values. Place the bounds values from the input
+                # granules in the corresponding places in the output bounds
+                # 2-D array. Also accumulate indices where bounds are applied,
+                # so that any missing bounds data can be identified.
+                input_indices = np.where(np.in1d(output_dimension_values,
+                                                 input_dimension.values))[0]
+
+                bounds_values[input_indices, :] = input_dimension.bounds_values[:]
+                filled_indices.update(input_indices)
+
+            if len(filled_indices) != output_dimension_values.size:
+                # The output bounds array has not been entirely filled
+                all_indices = set(range(output_dimension_values.size))
+                unfilled_indices = np.array(list(all_indices - filled_indices))
+                filled_indices = np.array(list(filled_indices))
+
+                # Find median differences between the input dimension values
+                # and the upper and lower bounds where the bounds are filled
+                median_lower = np.median(
+                    np.subtract(output_dimension_values,
+                                bounds_values.T[0])[filled_indices]
+                )
+                median_upper = np.median(
+                    np.subtract(bounds_values.T[1],
+                                output_dimension_values)[filled_indices]
+                )
+
+                # Apply the median differences to each point with missing
+                # bounds data to fill the rest of the bounds array.
+                bounds_values[unfilled_indices, 0] = np.subtract(
+                    output_dimension_values[unfilled_indices], median_lower
+                )
+                bounds_values[unfilled_indices, 1] = np.add(
+                    output_dimension_values[unfilled_indices], median_upper
+                )
+
+            # Remove any recurring decimal places:
```

Comment:
As with the aggregate output dimension values, the bounds of the longitude dimension for GPM/IMERG half hourly data were popping out recurring decimals. Turns out the `scale_to_integers` function has been more useful than first imagined.

Also, I let the precision for the bounds variables have one additional significant figure than the output dimension values. That's in case of situations such as:

Output dimension value 1 = 0.0
Output dimension value 2 = 0.5
Output bound between value 1 and value 2 = 0.25.